packages/core/src/hooks/useRawLogs.ts

```diff
@@ -1,10 +1,75 @@
-import { useEffect, useMemo, useState } from 'react'
+import { useEffect, useMemo, useRef, useState } from 'react'
 import { useEthers } from './useEthers'
 import { useReadonlyNetworks } from '../providers/network/readonlyNetworks'
-import { useBlockNumbers, useBlockNumber } from '../hooks'
+import { useBlockNumbers, useBlockNumber, useConfig } from '../hooks'
 import { QueryParams } from '../constants/type/QueryParams'
 import type { Filter, FilterByBlockHash, Log } from '@ethersproject/abstract-provider'
 import { Falsy } from '../model/types'
+import { ChainId } from '../constants'
+
+function deepEqual(obj1: any, obj2: any) {
+  if (obj1 === obj2)
+    // it's just the same object. No need to compare.
+    return true
+
+  if (obj1 == null) return obj1 == null
```

Comment:
```suggestion
  if (obj1 == null) return obj2 == null
```
I believe this is a typo

otherwise `deepEqual(null, 2)` would return true

Reply:
I don't know if this is intended behaviour  but it seems it would return 
`deepEqual(null, undefined)` as true

Reply:
Good catch!